### DISCUSSION TITLE
Allow beatmaps to have multiple switching rulesets +Video

### DISCUSSION THREAD

**Discussion by LIII-Works** [ppy/osu#13086] [Ideas]
**Describe the new feature:** 
-

> This feature allows the mapper to create a beatmap containing Osu!Standard, Osu!Mania, Osu!Taiko, Osu!Catch mode all in one

I made and attached a video to show you what I mean. Sorry for my bad editing. 

**Proposal designs of the feature:** 
-

-Soft momentary switching between gamemodes
  **_It should have a animation shows the elements exit from screen. Due to my bad editing I cant show you in the video._**

-Add a panel in beatmap editor to allow mapper to choose when to switch gamemode. Simply just insert different gamemode parts into timeline will be a good choice.

-Score caculation : Simply calculate with the formula of the original mode and then add up

**Attached Files**
-
[The video](https://streamable.com/khh2r)

**Comment by huoyaoyuan** [comment]
I recall a brainstorm around this in Chinese forum at about 2014.
The biggest problem of this idea will be **experience of difficulty**. Cross-ruleset players often have very different performance on each ruleset.
This may feel better with difficulty-streaming.
